expression/builtin_time_vec.go

```diff
@@ -731,11 +731,39 @@ func (b *builtinSubDurationAndDurationSig) vecEvalDuration(input *chunk.Chunk, r
 }
 
 func (b *builtinTimeToSecSig) vectorized() bool {
-	return false
+	return true
 }
 
+// evalInt evals TIME_TO_SEC(time).
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_time-to-sec
 func (b *builtinTimeToSecSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err = b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		var sign int
+		duration := buf.GetDuration(i, int(types.UnspecifiedFsp))
```

Comment:
Duration is a fixed-length type, we can use `buf.GoDurations` to get a slice, then we may not to deal with fsp in this case.

Reply:
here i need get `types.duration` type, because i need call `Hour` `Minute`... function. But `buf. GoDurations` get `times.duration` type slice

Reply:
Maybe we can export `types.splitDuration`.

Reply:
We really needn't `fsp` here. `times.Duration` has its similar method.  Foe example, https://golang.org/pkg/time/#Duration.Hours

Reply:
Maybe just [`Duration.Seconds()`](https://golang.org/pkg/time/#Duration.Seconds) works well.